reference/filone/reference.py

```
@@ -0,0 +1,319 @@
+        "productId": product["productId"],
+        "chainId": chain_id,
+        "pricingData": "0x" + pricing_bytes.hex(),
+        "termsSha256": hashlib.sha256(terms).hexdigest(),
+        "serviceOffer": offer,
+    }
+
+
+def validate_evidence(rendered, evidence):
+    exact_keys(evidence, EVIDENCE_KEYS, "evidence")
+    if type(evidence["schemaVersion"]) is not int or evidence["schemaVersion"] != 1:
+        raise ValueError("evidence.schemaVersion must be the integer 1")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)